eox_core/social_tpa_backends.py

```diff
@@ -55,6 +57,19 @@ def oidc_config(self):
             )
         return self.configuration
 
+    def get_user_details(self, response):
+        """
+        Override used to truncate first name in case is longer than the field's
+        length.
+        """
+        user_details = super().get_user_details(response)
+        first_name_max_len = User._meta.get_field("first_name").max_length  # pylint: disable=no-member, protected-access
+
+        if len(user_details.get("first_name", "")) > first_name_max_len:
```

Comment:
You don't even have to check length, if the stop value of the slice is greater than the original length it returns the whole thing.